### PR TITLE
Fix false positives in isLink and update test

### PR DESCRIPTION
**Pull request by AdamStone**
The isLink method (formerly isCID) appears [intended](https://github.com/ipld/js-ipld-dag-pb/blob/master/test/resolver.spec.js#L222) to identify paths specifically to link hashes (as opposed to paths to link name or size), but currently falls into the `true` case for paths to any truthy values. The tests pass because the non-valid CID case happens to check a path to an empty value.

Discussion [here](https://github.com/ipfs/js-ipfs/pull/1045#discussion_r159034560).

### CODE REVIEW

File: test/resolver.spec.js [hunk 1]
```diff
@@ -228,7 +228,7 @@ describe('IPLD Format resolver (local)', () => {
   })
 
   it('resolver.isLink for non valid CID', (done) => {
-    resolver.isLink(dataLinksNodeBlock, 'Links/0/Name', (err, link) => {
+    resolver.isLink(dataLinksNodeBlock, 'Links/0/Tsize', (err, link) => {
```
Comment (diasdavid) [inline test/resolver.spec.js]: Why the change in the test? It should also be false for /Name.

Comment (AdamStone) [reply test/resolver.spec.js]: True, but maybe it should check both then. Checking only name doesn't cover the general (non-blank-value) case.

**Comment by AdamStone** [comment]
@diasdavid I updated the test to check both Name and Size. I linked it and ran the ipld-resolver tests as well, and they also pass, but it seems like they don't actually hit this function.
